client/src/features/project/components/ProjectSettingsDescription.tsx

```diff
@@ -0,0 +1,158 @@
+import { useProjectMetadataQuery } from "../../projects/projectsKgApi";
+import { useUpdateDescriptionMutation } from "../projectCoreApi";
+import { CoreErrorAlert } from "../../../components/errors/CoreErrorAlert";
+import { CoreErrorContent } from "../../../utils/definitions";
+import { SettingRequiresKg } from "./ProjectSettingsUtils";
+
+interface ProjectSettingsDescriptionProps {
+  gitUrl: string;
+  isMaintainer: boolean;
+  projectFullPath: string;
+  projectId: number;
+}
+export function ProjectSettingsDescription({
+  gitUrl,
+  isMaintainer,
+  projectFullPath,
+  projectId,
+}: ProjectSettingsDescriptionProps) {
+  const [description, setDescription] = useState("");
+  const projectIndexingStatus = useGetProjectIndexingStatusQuery(projectId, {
+    skip: !projectFullPath || !projectId,
+  });
+  const projectMetadata = useProjectMetadataQuery(
+    { projectPath: projectFullPath, projectId },
+    {
+      skip:
+        !projectFullPath ||
+        !projectId ||
+        !projectIndexingStatus.data?.activated,
+    }
+  );
+  const [updateDescriptionMutation, updateDescriptionStatus] =
+    useUpdateDescriptionMutation();
+  const onSubmit = useCallback(() => {
+    updateDescriptionMutation({
+      description,
+      gitUrl,
+      projectId,
+    });
+  }, [description, gitUrl, projectId, updateDescriptionMutation]);
+
+  const setDescriptionAndReset = (newDescription: string) => {
+    setDescription(newDescription);
+    // Reset mutation when changing description after an update.
+    if (!updateDescriptionStatus.isUninitialized)
+      updateDescriptionStatus.reset();
+  };
+
+  useEffect(() => {
+    if (projectMetadata.data?.description)
+      setDescription(projectMetadata.data?.description);
+  }, [projectMetadata.data?.description]);
+
+  const readOnly =
+    !isMaintainer ||
+    projectIndexingStatus.isLoading ||
+    projectMetadata.isLoading;
+  const pristine = description === projectMetadata.data?.description;
+
+  if (projectIndexingStatus.isLoading || projectMetadata.isLoading)
+    return (
+      <DescriptionCard>
+        <Loader className="ms-1" inline size={16} />
+      </DescriptionCard>
+    );
+
+  if (projectIndexingStatus.data?.activated === false)
+    return (
+      <DescriptionCard>
+        <SettingRequiresKg />
+      </DescriptionCard>
+    );
+
+  const inputField = (
+    <Input
+      data-cy="description-input"
+      disabled={updateDescriptionStatus.isLoading}
+      id="projectDescription"
+      onChange={(e) => setDescriptionAndReset(e.target.value)}
+      readOnly={readOnly}
+      value={description}
+    />
+  );
+
+  const submitButton = readOnly ? null : (
+    <InlineSubmitButton
+      className="updateProjectSettings"
+      doneText="Updated"
+      id="update-desc"
+      isDone={updateDescriptionStatus.isSuccess}
+      isReadOnly={readOnly || pristine}
+      isSubmitting={updateDescriptionStatus.isLoading}
+      onSubmit={() => onSubmit()}
```

Comment:
Technicality: `() => onSubmit()` creates a new function at each render, while `onSubmit` may stay the same.

```suggestion
      onSubmit={onSubmit}
```